server/tests/order/test_schemas.py

```diff
@@ -0,0 +1,79 @@
+"""Test for order schema serialization."""
+
+from datetime import UTC, datetime
+from uuid import UUID
+
+import pytest
+
+from polar.models.order import (
+    OrderBillingReason,
+    OrderBillingReasonInternal,
+    OrderStatus,
+)
+from polar.order.schemas import OrderBase
+
+# Mapping of all internal billing reasons to their expected serialized output
+BILLING_REASON_SERIALIZATION_MAP = {
+    OrderBillingReasonInternal.purchase: OrderBillingReason.purchase,
+    OrderBillingReasonInternal.subscription_create: OrderBillingReason.subscription_create,
+    OrderBillingReasonInternal.subscription_cycle: OrderBillingReason.subscription_cycle,
+    OrderBillingReasonInternal.subscription_cycle_after_trial: OrderBillingReason.subscription_cycle,
+    OrderBillingReasonInternal.subscription_update: OrderBillingReason.subscription_update,
+}
+
+# Ensure the mapping is exhaustive for all internal billing reasons
+assert set(BILLING_REASON_SERIALIZATION_MAP.keys()) == set(
+    OrderBillingReasonInternal
+), "BILLING_REASON_SERIALIZATION_MAP must cover all OrderBillingReasonInternal values"
+
+
+class TestOrderBillingReasonSerializer:
+    """Test OrderBase billing_reason field serializer."""
+
+    @pytest.mark.parametrize(
+        "internal_reason,expected_output",
+        list(BILLING_REASON_SERIALIZATION_MAP.items()),
+    )
+    def test_billing_reason_serialization(
+        self,
+        internal_reason: OrderBillingReasonInternal,
+        expected_output: OrderBillingReason,
+    ) -> None:
+        """Test that billing reasons are serialized correctly."""
+        # Use model_construct to bypass validation
+        test_uuid = UUID("12345678-1234-5678-1234-567812345678")
```

Comment:
Another approach could be to create an Order fixture, validate it as the Pydantic schema and then test the output:

```py
order = await create_order(
        save_fixture,
        product=product,
        customer=customer,
        billing_reason=internal_reason,
    )

order_schema = Order.model_validate(order)
serialized = order.model_dump(mode="json")
assert serialized["billing_reason"] == expected_output
```